stock_replenishment_mrp_bom_selection/wizard/stock_warehouse_orderpoint_replenish_wizard.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2024 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from datetime import datetime
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class StockWarehouseOrderpointReplenishWizard(models.TransientModel):
+    _name = "stock.warehouse.orderpoint.replenish.wizard"
+    _description = "Stock warehouse orderpoint replenish wizard"
+
+    orderpoint_id = fields.Many2one("stock.warehouse.orderpoint")
+    qty_to_order = fields.Float(related="orderpoint_id.qty_to_order")
+    product_id = fields.Many2one("product.product", related="orderpoint_id.product_id")
+    product_tmpl_id = fields.Many2one(
+        "product.template", related="product_id.product_tmpl_id"
+    )
+    bom_ids = fields.One2many("mrp.bom", related="product_tmpl_id.bom_ids")
+    bom_line_ids = fields.One2many(
+        comodel_name="bom.line.wizard",
+        inverse_name="wizard_id",
+        compute="_compute_bom_line_ids",
+        store=True,
+        readonly=False,
+    )
+
+    @api.depends("orderpoint_id")
+    def _compute_bom_line_ids(self):
+        self.bom_line_ids = None
+        lines_vals = []
+        for line in self.bom_ids:
+            lines_vals.append(
+                (
+                    0,
+                    0,
+                    {
+                        "wizard_id": self.id,
+                        "bom_id": line.id,
+                    },
+                )
+            )
+        self.bom_line_ids = lines_vals
+
+    def action_confirm(self):
+        datetime.now()
+        total_qty_to_produce = sum(self.bom_line_ids.mapped("qty_to_produce"))
+        if total_qty_to_produce > self.qty_to_order:
+            raise UserError(
+                _("The quantities to be produced exceed the quantity to be ordered.")
+            )
+        if total_qty_to_produce == 0:
+            raise UserError(_("Nothing to produce."))
+        orderpoint = self.env["stock.warehouse.orderpoint"].browse(
+            self.orderpoint_id.id
+        )
+        initial_qty_to_order = orderpoint.qty_to_order
```

Comment:
Why using this if you already have it in a field in the wizard?

Reply:
Because self.qty_to_order is related to orderpoint_id.qty_to_order. When the ‘_procure_orderpoint_confirm’ method is called, which is the one that executes the standard logic, it sets orderpoint_id.qty_to_order to 0. That's why I use a for, to call the standard method passing the quantity and the bill of materials and then retrieving the initial quantity and recalculating if not everything has been replenished or leaving a negative value in case it has been over manufactured. To avoid negative values you can restrict the maximum quantity to be ordered. In any case the calculation is done correctly.